scripts/zones/Northern_San_dOria/npcs/Charlaimagnat.lua

```diff
@@ -32,7 +33,21 @@ end;
 -----------------------------------
 
 function onTrigger(player,npc)
-	player:startEvent(0x02be);
+
+	local realday = tonumber(tostring(os.date("%Y")) .. os.date("%m") .. os.date("%d"));
+	local TheMissingPiece = player:getQuestStatus(OUTLANDS,THE_MISSING_PIECE);
+	
+	if(TheMissingPiece == QUEST_ACCEPTED and player:hasKeyItem(TABLET_OF_ANCIENT_MAGIC) and player:hasKeyItem(LETTER_FROM_ALFESAR)) then
+		player:startEvent(0x02bf);--Continuing the Quest
+		player:setVar("TheMissingPiece_date", 0);

```

Comment:
Why do you set it to zero here, but in onEventFinish you set it again? I'd just delete this line